src/geocoding/geocoding.py

```diff
@@ -290,40 +285,31 @@ def is_valid_address(street, number, locality, region, country):
     return True, "OK"
 
 
-def _get_polygon_geojson(osm_id, osm_type):
-    """
-    :param int osm_id: see https://wiki.openstreetmap.org/wiki/Elements
-    :param char osm_type: N | W | R
-    :return: dict (geojson of the requested element).
-    """
-    params = {
-        'format': 'json',
-        'osm_id': osm_id,
-        'osm_type': osm_type,
-        'polygon_geojson': 1,
-    }
-    url = "http://nominatim.openstreetmap.org/reverse"
-    r = requests.get(url, params=params)
-    if r.ok:
-        return r.json().get('geojson', None)
+async def geocode(key: str):
+    async with Nominatim(
+            user_agent="quantumleap",
+            adapter_factory=AioHTTPAdapter,
+    ) as geolocator:
+        info = await geolocator.geocode(key, geometry='geojson', exactly_one=False, limit=5, timeout=30)
+        return info
```

Comment:
@chicco785 you sure you want to swap out the sync API for the async one? I'm not convinced in general we're going to get major performance benefits without changing the current QL architecture. Isn't this code just a case in point? i.e. we're using the async API but then in practice the await statement makes the call synchronous just like we had it before. Am I missing something here?

Reply:
@c0c0n3 i think it would be ideal to go async with all external calls, this would be just a first step

Reply:
Yeah, I like the idea, but I'm a bit scared about the effort we'll have to put in. Like I said, I don't think we're gonna be able to improve performance significantly without changing the architecture. You can read about why I think that [over here](https://github.com/orchestracities/ngsi-timeseries-api/wiki/No-async-free-Lunch) and also change this code accordingly if you have time, otherwise we can just leave it be, we can refactor later.

Reply:
@c0c0n3 given that the introduced overhead maybe negligible (in your measurements is 0.2%), i think a gradual approach makes sense. also in this specific case, considering your comments about avoiding duplicated calls, we already use caching, i.e. we call the geocode only if the values are not in the cache.

in general, I believe it would be good to think if there are things that can be run in parallel, but that only as a second step. for example, instead of adding the geo-resolution of data immediately when injecting data, we could do that in a second step. in general this could apply to any secondary data manipulation, and could be also the way to parallelize inserts of batches.

Reply:
sure, if there's any easy wins we should go for it! re: the overhead, there's no overhead if we go async---sorry if that wasn't clear in the wiki page I linked, I wrote that too quickly I guess. The point is the code we have here should change if you want to the `geocode` calls to run concurrently at the moment they get serialised---e.g. by using `gather` as explained in the wiki page.